### PR TITLE
Save list of custom packages

### PR DESCRIPTION
**Pull request by Saurabh-Thakre**
# Description

Functionality for images endpoint to accept the customer provided list of custom packages from the front end and saves this list

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/images.go [hunk 1]
```diff
@@ -1079,6 +1079,11 @@ func (s *ImageService) GetUpdateInfo(image models.Image) ([]models.ImageUpdateAv
 			s.log.WithField("error", err.Error()).Error("Error retrieving updated packages")
 			return nil, err
 		}
+
+		if err := db.DB.Model(&upd).Association("CustomPackages").Find(&upd.CustomPackages); err != nil {
```
Comment (ldjebran) [inline pkg/services/images.go:1083]: If we are adding the custom packages here may be we should also update the GetDiffOnUpdate function to count the custom repos  in added removed packages also ?

File: pkg/models/images.go [hunk 2]
```diff
@@ -185,3 +186,18 @@ func checkIfImageExist(imageName string) bool {
 	}
 	return imageFindByName != nil
 }
+
+// GetALLPackagesList returns all the packages including custom packages containing their names
+func (i *Image) GetALLPackagesList() *[]string {
```
Comment (ldjebran) [inline pkg/models/images.go:191]: may be like this 
```go
func (i *Image) GetALLPackagesList() *[]string {
	initialPackages := *i.GetPackagesList()
	packages := make([]string, 0, len(initialPackages)+len(i.CustomPackages))

	for _, pkg := range initialPackages {
		packages = append(packages, pkg)
	}

	for _, pkg := range i.CustomPackages {
		packages = append(packages, pkg)
	}
	return &packages
}
```

**Review by ldjebran — APPROVED** [review]
Overall OK remain two comments about updating GetDiffOnUpdate  and GetALLPackagesList functions and some unittests for GetALLPackagesList.

File: pkg/services/images.go (hunk 1 above)
Comment (Saurabh-Thakre) [reply pkg/services/images.go:1089]: count the custom **repos** 
> Did you mean to count custom packages ?

**Pull request merged**